tests/conftest.py

```diff
@@ -116,3 +133,109 @@ def structurally_valid_vote():
             'timestamp': '1111111111'
         }
     }
+
+
+@pytest.fixture
+def setup_database(request, restore_config, node_config):
+    from bigchaindb.backend import connect, schema
+    from bigchaindb.common.exceptions import DatabaseAlreadyExists
+    print('Initializing test db')
+    db_name = node_config['database']['name']
+    conn = connect()
+
+    if conn.run(r.db_list().contains(db_name)):
+        conn.run(r.db_drop(db_name))
+
+    try:
+        schema.init_database()
+    except DatabaseAlreadyExists:
+        print('Database already exists.')
+
+    print('Finishing init database')
+
+    def fin():
+        print('Deleting `{}` database'.format(db_name))
+        conn = connect()
+        try:
+            conn.run(r.db_drop(db_name))
+        except r.ReqlOpFailedError as e:
+            if e.message != 'Database `{}` does not exist.'.format(db_name):
+                raise
+        print('Finished deleting `{}`'.format(db_name))
+
+    request.addfinalizer(fin)
+
+
+@pytest.fixture
+def cleanup_tables(request, node_config):
+    db_name = node_config['database']['name']
+
+    def fin():
+        from bigchaindb.backend import connect
+        conn = connect()
+        try:
+            conn.run(r.db(db_name).table('bigchain').delete())
+            conn.run(r.db(db_name).table('backlog').delete())
+            conn.run(r.db(db_name).table('votes').delete())
+        except r.ReqlOpFailedError as e:
+            if e.message != 'Database `{}` does not exist.'.format(db_name):
+                raise
+
+    request.addfinalizer(fin)
+
+
+@pytest.fixture
+def inputs(user_pk, cleanup_tables, setup_database):
+    from bigchaindb import Bigchain
+    from bigchaindb.models import Transaction
+    from bigchaindb.common.exceptions import GenesisBlockAlreadyExistsError
+    # 1. create the genesis block
+    b = Bigchain()
+    try:
+        g = b.create_genesis_block()
+    except GenesisBlockAlreadyExistsError:
+        pass
+
+    # 2. create blocks with transactions for `USER` to spend
+    prev_block_id = g.id
+    for block in range(4):
+        transactions = [
+            Transaction.create([b.me], [([user_pk], 1)]).sign([b.me_private])
+            for i in range(10)
+        ]
+        block = b.create_block(transactions)
+        b.write_block(block)
+
+        # 3. vote the blocks valid, so that the inputs are valid
+        vote = b.vote(block.id, prev_block_id, True)
+        prev_block_id = block.id
+        b.write_vote(vote)
+
+
+@pytest.fixture
+def inputs_shared(user_pk, user2_pk):
```

Comment:
Just a note for myself: this will need to be updated in https://github.com/bigchaindb/bigchaindb/pull/926.